packages/language-server/src/plugins/typescript/service.ts

```diff
@@ -109,30 +106,36 @@ export function createLanguageService(
     };
 
     function updateDocument(document: TypescriptDocument): ts.LanguageService {
-        const preSnapshot = documents.get(document.getFilePath()!);
+        const preSnapshot = snapshotManager.get(document.getFilePath()!);
         const newSnapshot = DocumentSnapshot.fromDocument(document);
         if (preSnapshot && preSnapshot.scriptKind !== newSnapshot.scriptKind) {
             // Restart language service as it doesn't handle script kind changes.
             languageService.dispose();
             languageService = ts.createLanguageService(host);
         }
 
-        documents.set(document.getFilePath()!, newSnapshot);
+        snapshotManager.set(document.getFilePath()!, newSnapshot);
         return languageService;
     }
 
+    function getScriptSnapshot(fileName: string): DocumentSnapshot | undefined {
+        return getSvelteSnapshot(fileName) ?? snapshotManager.get(fileName);
+    }
+
     function getSvelteSnapshot(fileName: string): DocumentSnapshot | undefined {
+        if (!isSvelteFilePath(fileName)) {
+            return
+        }
+
         fileName = ensureRealSvelteFilePath(fileName);
```

Comment:
`ensureRealSvelteFilePath` has to be called before the `if`-check, else the check always returns false
```suggestion
        fileName = ensureRealSvelteFilePath(fileName);

        if (!isSvelteFilePath(fileName)) {
            return
        }
```